Source/Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func/PreparingToSend/PreparingToSendOrchestration.cs

```diff
@@ -0,0 +1,287 @@
+        /// <param name="processRecipientDataListActivity">Process recipient data list activity.</param>
+        /// <param name="createSendingNotificationActivity">Create sending notification activity.</param>
+        /// <param name="sendTriggersToSendFunctionActivity">Send triggers to send function sub-orchestration.</param>
+        /// <param name="notificationDataRepository">The notification data repository.</param>
+        /// <param name="handleFailureActivity">Clean up activity.</param>
+        public PreparingToSendOrchestration(
+            GetRecipientDataListForAllUsersActivity getRecipientDataListForAllUsersActivity,
+            GetTeamDataEntitiesByIdsActivity getTeamDataEntitiesByIdsActivity,
+            GetRecipientDataListForRosterActivity getRecipientDataListForRosterActivity,
+            GetRecipientDataListForTeamsActivity getRecipientDataListForTeamsActivity,
+            ProcessRecipientDataListActivity processRecipientDataListActivity,
+            CreateSendingNotificationActivity createSendingNotificationActivity,
+            SendTriggersToSendFunctionActivity sendTriggersToSendFunctionActivity,
+            NotificationDataRepository notificationDataRepository,
+            HandleFailureActivity handleFailureActivity)
+        {
+            this.getRecipientDataListForAllUsersActivity = getRecipientDataListForAllUsersActivity;
+            this.getTeamDataEntitiesByIdsActivity = getTeamDataEntitiesByIdsActivity;
+            this.getRecipientDataListForRosterActivity = getRecipientDataListForRosterActivity;
+            this.getRecipientDataListForTeamsActivity = getRecipientDataListForTeamsActivity;
+            this.processRecipientDataListActivity = processRecipientDataListActivity;
+            this.createSendingNotificationActivity = createSendingNotificationActivity;
+            this.sendTriggersToSendFunctionActivity = sendTriggersToSendFunctionActivity;
+            this.notificationDataRepository = notificationDataRepository;
+            this.handleFailureActivity = handleFailureActivity;
+        }
+
+        /// <summary>
+        /// This is the durable orchestration method,
+        /// which kicks of the preparing to send process.
+        /// </summary>
+        /// <param name="context">Durable orchestration context.</param>
+        /// <param name="log">Logging service.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        [FunctionName(nameof(PrepareToSendOrchestrationAsync))]
+        public async Task PrepareToSendOrchestrationAsync(
+            [OrchestrationTrigger] DurableOrchestrationContext context,
+            ILogger log)
+        {
+            var notificationDataEntity = context.GetInput<NotificationDataEntity>();
+
+            if (!context.IsReplaying)
+            {
+                log.LogCritical($"Start to prepare to send the notification {notificationDataEntity.Id}!");
```

Comment:
Lower this. Critical log level is for "Logs that describe an unrecoverable application or system crash, or a catastrophic failure that requires immediate attention."

https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loglevel?view=dotnet-plat-ext-3.1#fields